ApollosProject/apollos-prototype
# Handle Links in the HTMLView.

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds a prop to the HTMLView which can be used to pass a link handler. This will allow a partner church to choose how they would like to handle links (anchors) in the HTMLView. The default is react-native Linking. The default for Apollos Core is the in-app browser

### What design trade-offs/decisions were made?

### How do I test this PR?

1. Open an article with an anchor
2. Tap the anchor; it should properly route you, using the in-app browser
3. Comment out ln 25 of `/packages/apolloschurchapp/src/content-single/HTMLContent/index.js`
4. Open the same article
5. Tap the anchor, it should properly route you, using react-native Linking

### What automated tests did you write?

I did not add in anything regarding the HTMLView stories as this would require us to use the in-app browser as a dependency for the HTMLView package, which we were trying to avoid.

## SCREENSHOTS

|Linking|In-App|
|---|---|
|![Linking](https://user-images.githubusercontent.com/29125070/69067273-cc250280-09f0-11ea-9bee-c29fc8670695.gif)|![In-App](https://user-images.githubusercontent.com/29125070/69067288-d0512000-09f0-11ea-9175-87d406830039.gif)|

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
